### PR TITLE
deps: bump optimize engine dep and remove Optimize license deps

### PR DESCRIPTION
**Pull request by HeleneW-dot**
## Description

This is to resolve the issue reported [here](https://camunda.slack.com/archives/C06HTSPD5AP/p1723205084816939): `org.camunda.bpm.model:camunda-dmn-model:jar:7.19.11-ee` cannot be resolved, bumping to 7.21.0 instead. 

`org.camunda.bpm:camunda-license-check` can also not be accessed, so this PR removes the Optimize logic requiring this dependency.
I am not removing CAMUNDA_OPTIMIZE_ENTERPRISE, the related config and its usage in the UIConfigurationResponse, follow up to handle this [here](https://github.com/camunda/camunda-optimize/issues/14022)

Also adjusts the optimize BE jetty version to mitigate failing Zeebe IT. 

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: optimize/backend/src/main/java/io/camunda/optimize/rest/security/platform/PlatformSecurityConfigurerAdapter.java [hunk 1]
```diff
@@ -146,7 +145,6 @@ protected SecurityFilterChain configureWebSecurity(HttpSecurity http) {
                         new AntPathRequestMatcher(createApiPath(STATUS_PATH)),
                         new AntPathRequestMatcher(createApiPath(UI_CONFIGURATION_PATH)),
                         new AntPathRequestMatcher(createApiPath(LOCALIZATION_PATH)),
-                        new AntPathRequestMatcher(createApiPath(LICENSE_PATH, SUB_PATH_ANY)),
                         new AntPathRequestMatcher(createApiPath(AUTHENTICATION_PATH)))
```
Comment (RomanJRW) [inline optimize/backend/src/main/java/io/camunda/optimize/rest/security/platform/PlatformSecurityConfigurerAdapter.java:148]: `OptimizeRequestExecutor#buildValidateLicenseRequest` can be deleted

File: optimize/backend/src/main/java/io/camunda/optimize/rest/security/platform/PlatformSecurityConfigurerAdapter.java [hunk 2]
```diff
@@ -146,7 +145,6 @@ protected SecurityFilterChain configureWebSecurity(HttpSecurity http) {
...
                     .permitAll()
```
Comment (RomanJRW) [inline optimize/backend/src/main/java/io/camunda/optimize/rest/security/platform/PlatformSecurityConfigurerAdapter.java:149]: `OptimizeRequestExecutor#buildValidateAndStoreLicenseRequest` can be deleted

File: optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/query/LicenseInformationResponseDto.java [hunk 3]
```diff
@@ -1,21 +0,0 @@
...
- */
-package io.camunda.optimize.dto.optimize.query;
-
-import io.camunda.optimize.dto.optimize.OptimizeDto;
-import java.io.Serializable;
-import java.time.OffsetDateTime;
-import lombok.Data;
-
-@Data
-public class LicenseInformationResponseDto implements OptimizeDto, Serializable {
```
Comment (RomanJRW) [inline optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/query/LicenseInformationResponseDto.java:16]: did you update the REST API docs for this removal too?

**Review by RomanJRW — APPROVED** [review]
LGTM ✅

File: optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/query/LicenseInformationResponseDto.java (hunk 3 above)
Comment (PHWaechtler) [reply optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/query/LicenseInformationResponseDto.java:16]: Tragically this was the first time I remembered, yet there is no docs for this on confluence

File: optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/AbstractUpgradeIT.java.out [hunk 4]
```diff
@@ -6,348 +6,348 @@
...
+//import static io.camunda.optimize.upgrade.es.SchemaUpgradeClientFactory.createSchemaUpgradeClient;
+//import static jakarta.ws.rs.HttpMethod.DELETE;
+//import static jakarta.ws.rs.HttpMethod.POST;
+//import static org.elasticsearch.index.query.QueryBuilders.matchAllQuery;
+//import static org.elasticsearch.index.query.QueryBuilders.termQuery;
+//import static org.mockserver.model.HttpRequest.request;
+//
+//import com.fasterxml.jackson.databind.ObjectMapper;
+//import io.camunda.optimize.exception.OptimizeIntegrationTestException;
+//import io.camunda.optimize.service.db.DatabaseConstants;
```
Comment (Zelldon) [inline optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/AbstractUpgradeIT.java.out:23]: ❓ Do we need the class outcommented? You can also find this via git history no need to keep such files. :)

File: optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/migrate313to86/DeleteLicenseIndex86IT.java.out [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+// TODO recreate C8 IT equivalent of this with #13337
+//package io.camunda.optimize.upgrade.migrate313to86;
+//
+//import static org.assertj.core.api.Assertions.assertThat;
+//
+//import java.util.List;
+//import lombok.SneakyThrows;
+//import org.junit.jupiter.api.Test;
+//
+//public class DeleteLicenseIndex86IT extends AbstractUpgrade86IT {
```
Comment (Zelldon) [inline optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/migrate313to86/DeleteLicenseIndex86IT.java.out:17]: 🔧 Same here :)

**Review by ChrisKujawa — APPROVED** [review]
Love it. Deleted code is good code 😍 :)

File: optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/AbstractUpgradeIT.java.out (hunk 4 above)
Comment (PHWaechtler) [reply optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/AbstractUpgradeIT.java.out:23]: Yeah, we've got a whole bunch of IT in Optimize that are currently C7 only but which we are planning to migrate to C8. We will be working our way through all of these commented out files and add them back in as shiny new C8 IT.

**Pull request merged**